server/mscalendar/auto_respond_test.go

```diff
@@ -0,0 +1,169 @@
+			recipientActiveEvents:      []string{},
+			recipientStatusString:      model.STATUS_DND,
+			autoRespondSetting:         true,
+			autoRespondMessageSetting:  "Hello, I'm in a meeting and will respond to your message as soon as I'm free.",
+			expectedDMMessage:          "",
+		},{
+			name:                       "Recipient autorespond Setting turned off",
+			expectedError:              "",
+			recipientActiveEvents:      []string{"active_event_hash"},
+			recipientStatusString:      model.STATUS_DND,
+			autoRespondSetting:         false,
+			autoRespondMessageSetting:  "Hello, I'm in a meeting and will respond to your message as soon as I'm free.",
+			expectedDMMessage:          "",
+		},{
+			name:                       "Recipient user status is set to online",
+			expectedError:              "",
+			recipientActiveEvents:      []string{"active_event_hash"},
+			recipientStatusString:      model.STATUS_ONLINE,
+			autoRespondSetting:         true,
+			autoRespondMessageSetting:  "Hello, I'm in a meeting and will respond to your message as soon as I'm free.",
+			expectedDMMessage:          "",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := mock_store.NewMockStore(ctrl)
+			mockPoster := mock_bot.NewMockPoster(ctrl)
+			mockRemote := mock_remote.NewMockRemote(ctrl)
+			mockPluginAPI := mock_plugin_api.NewMockPluginAPI(ctrl)
+
+			conf := &config.Config{PluginVersion: "x.x.x"}
+			env := Env{
+				Config: conf,
+				Dependencies: &Dependencies{
+					Store:     mockStore,
+					Logger:    &bot.NilLogger{},
+					Poster:    mockPoster,
+					Remote:    mockRemote,
+					PluginAPI: mockPluginAPI,
+				},
+			}
+
+			post := newTestMattermostPost()
+			channel := newTestMattermostChannel()
+			mattermostUserSender := newTestMattermostUser("mattermost_user_sender_id")
+			mattermostUserRecipient := newTestMattermostUser("mattermost_user_recipient_id")
+
+			recipientStatus := newTestMattermostStatus(tc.recipientStatusString)
+			storedRecipient := newTestStoreUser(tc.recipientActiveEvents, "mattermost_user_recipient_id", tc.autoRespondSetting, tc.autoRespondMessageSetting)
+
+			mockPluginAPI.EXPECT().GetMattermostChannel("mattermost_post_channel_id").Return(channel, nil)
+			mockPluginAPI.EXPECT().GetMattermostUsersInChannel("mattermost_post_channel_id", model.CHANNEL_SORT_BY_USERNAME, 0, 2).Return([]*model.User{mattermostUserSender, mattermostUserRecipient}, nil)
+
+			mockStore.EXPECT().LoadUser("mattermost_user_sender_id").Return(nil, errors.New("user not found"))
+			mockStore.EXPECT().LoadUser("mattermost_user_recipient_id").Return(storedRecipient, nil)
+
+			if tc.autoRespondSetting && len(tc.recipientActiveEvents) > 0 {
+				mockPluginAPI.EXPECT().GetMattermostUserStatus("mattermost_user_recipient_id").Return(recipientStatus, nil)
+			}
+
+			if tc.recipientStatusString != model.STATUS_ONLINE && tc.expectedDMMessage != "" {
```

Comment:
This way we are not assuming any logic of how the source code works

```suggestion
			if tc.expectedDMMessage != "" {
```